### PR TITLE
[Type checker] Provide actual ObjC runtime name in @NSKeyedArchiveLegacy Fix-Its.

### PR DESCRIPTION
**Pull request by DougGregor**
While here, ban users from ever uttering `@ _staticInitializeObjCMetadata`.

Fixes rdar://problem/32118614.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

File: lib/Sema/TypeCheckProtocol.cpp [hunk 1]
```diff
@@ -6048,13 +6049,16 @@ void TypeChecker::checkConformancesInContext(DeclContext *dc,
           auto insertionLoc =
             classDecl->getAttributeInsertionLoc(/*forModifier=*/false);
           if (isFixable) {
+            Mangle::ASTMangler mangler;
```
Comment (jrose-apple) [inline lib/Sema/TypeCheckProtocol.cpp]: Can you leave behind a comment noting that this is deliberately using the Swift 3 mangling?

Comment (DougGregor) [inline lib/Sema/TypeCheckProtocol.cpp]: Sure.

**Review by jrose-apple — APPROVED** [review]
Thanks, Doug!

Erik and I also noticed another inconsistency, which is that we shouldn't be doing any of this on Linux Foundation. (Neither the fix-its nor the eager initialization will work, right?) But at least no one will be using it there, because that already fails to handle non-simple mangled names.

**Comment by DougGregor** [comment]
yeah, got it https://github.com/apple/swift/pull/9487

**Pull request merged**